rln-js/index.html

```diff
@@ -176,6 +188,118 @@
     }
 
+
+    const rlnDeployBlk = 7109391;
+    const rlnAddress = "0x4252105670fe33d2947e8ead304969849e64f2a6";
+    const rlnAbi = [
+        "function MEMBERSHIP_DEPOSIT() public view returns(uint256)",
+        "function register(uint256 pubkey) external payable",
+        "function withdraw(uint256 secret, uint256 _pubkeyIndex, address payable receiver) external",
+        "event MemberRegistered(uint256 pubkey, uint256 index)",
+        "event MemberWithdrawn(uint256 pubkey, uint256 index)"
+    ];
+
+    const provider = new ethers.providers.Web3Provider(window.ethereum);
+
+    let accounts;
+    let rlnContract;
+
+    const handleMembership = (pubkey, index, event) => {
+        try {
+            const idCommitment = ethers.utils.zeroPad(ethers.utils.arrayify(pubkey), 32);
+            rlnInstance.insertMember(idCommitment);
+            console.debug("IDCommitment registered in tree", idCommitment, index.toNumber());
+        } catch(err){
+            alert(err); // TODO: the merkle tree can be in a wrong state. The app should be disabled
+        }
+    }
+
+    const setupRLN = async (provider) => {
+        rlnContract = new ethers.Contract(rlnAddress, rlnAbi, provider);
+        
+        const filter = rlnContract.filters.MemberRegistered()
+
+        // populating merkle tree:
+        const alreadyRegisteredMembers = await rlnContract.queryFilter(filter, rlnDeployBlk)
+        alreadyRegisteredMembers.forEach(event => {
+            handleMembership(event.args.pubkey, event.args.index, event);
+        });
+
+        alert("Loaded all events from contract");
+
+        // TODO: at this point it should be possible to register a credential and send messages
+        // TODO: emit an event to enable message form and register button
+
+        // reacting to new registrations
+        rlnContract.on(filter, (pubkey, index, event) => {
+            handleMembership(event.args.pubkey, event.args.index, event);
+        });
+    }
+
+    const setAccounts = acc => {
+        accounts = acc;
+        walletDiv.innerHTML = accounts.length ? accounts[0] : "";
+    }
+
+    const checkChain = async (chainId) => {
+        if(chainId == 5){
+            setupRLN(provider);
+            connectWalletButton.disabled = false;
+        } else {
+            alert("Switch to Goerli");
+            connectWalletButton.disabled = true;
+        }
+    }
+
+    connectWalletButton.onclick = async () => {
+        try {
+            accounts = await provider.send("eth_requestAccounts", []);
+            setAccounts(accounts);
+
+            const network = await provider.getNetwork();
+            checkChain(network.chainId);
+        } catch (e) {
+            ("No web3 provider available", e);
+        }
+    };
+
+    window.ethereum.on('accountsChanged', setAccounts);
+    window.ethereum.on('chainChanged', chainId => {
+        checkChain(parseInt(chainId, 16));
+    });
+
+
+    const registerCredentials = async (idCommitment) => {
```

Comment:
I woudl not bother do a separate function if it's only used in one place. just set it directly to registerButton.onclick